### PR TITLE
Improve database query performance

### PR DESCRIPTION
**Pull request by xq262144**
1. Add trace regions and trace wrapper.
2. Defer response signing process.
3. Log fetch logic for follower live recover.

### CODE REVIEW

File: rpc/clientpool_caller.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	case *Client:
+		client = v
+	case error:
+		err = v
+	}
+
+	return
+}
+
+func (c *ClientPoolCaller) Call(method string, args interface{}, reply interface{}) (err error) {
```
Comment (covenantdb) [inline rpc/clientpool_caller.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method ClientPoolCaller.Call should have comment or be unexported

File: rpc/clientpool_caller.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	if err != nil {
+		err = errors.Wrapf(err, "call %s failed", method)
+	}
+
+	return
+}
+
+func (c *ClientPoolCaller) Close() {
```
Comment (covenantdb) [inline rpc/clientpool_caller.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method ClientPoolCaller.Close should have comment or be unexported

File: rpc/pool.go [hunk 3]
```diff
@@ -20,17 +20,16 @@ import (
...
 	"sync"
 
+	"github.com/CovenantSQL/CovenantSQL/proto"
 	"github.com/pkg/errors"
 	mux "github.com/xtaci/smux"
-
-	"github.com/CovenantSQL/CovenantSQL/proto"
 )
 
+const MaxPhysicalConnection = 10
```
Comment (covenantdb) [inline rpc/pool.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported const MaxPhysicalConnection should have comment or be unexported

File: alltest.sh [hunk 4]
```diff
@@ -4,30 +4,15 @@ set -o errexit
...
-
 main() {
   make clean
   make -j6 bp miner observer
 
-  # test package by package
-  for package in $(go list ./... | grep -v "/vendor/"); do
-    test::package "${package}"
-  done
+  go test -tags "$UNITTESTTAGS" -race -failfast -parallel 16 -cpu 16 -coverprofile main.cover.out $(go list ./... | grep -v CovenantSQL/api)
```
Comment (ggicci) [inline alltest.sh:11]: Shall we exclude `vendor` folders?

Comment (xq262144) [reply alltest.sh:11]: `vendor` folder is already excluded by `go list` or `go test` command.

**Review by ggicci — APPROVED** [review]

**Review by auxten — APPROVED** [review]
LGTM, ship it!

**Review by zeqing-guo — APPROVED** [review]
Approve

**Pull request merged**